core/iwasm/interpreter/wasm_loader.c

```diff
@@ -11590,6 +11618,61 @@ wasm_loader_prepare_bytecode(WASMModule *module, WASMFunction *func,
                         break;
                     }
 
+#if WASM_ENABLE_GC_BINARYEN != 0
+                    case WASM_OP_ARRAY_COPY:
+                    {
+                        uint32 src_type_idx;
+                        WASMRefType src_ref_type = { 0 };
+                        WASMRefType dst_ref_type = { 0 };
+                        WASMArrayType *array_type;
+                        /* typeidx1 */
+                        read_leb_uint32(p, p_end, type_idx);
+#if WASM_ENABLE_FAST_INTERP != 0
+                        emit_uint32(loader_ctx, type_idx);
+#endif
+                        /* typeidx2 */
+                        read_leb_uint32(p, p_end, src_type_idx);
+#if WASM_ENABLE_FAST_INTERP != 0
+                        emit_uint32(loader_ctx, src_type_idx);
+#endif
```

Comment:
Should check type_idx and src_type_idx, like:
```C
                    if (!check_type_index(module, type_idx, error_buf,
                                          error_buf_size)
                       || check_type_index(module, src_type_idx, error_buf,
                                          error_buf_size)) {
                        goto fail;
                    }
```
And check whether they are array type:
```
if (module->types[type_idx]->type_flag != WASM_TYPE_ARRAY
    || module->types[src_type_idx]->type_flag != WASM_TYPE_ARRAY) {
    set_error_buf(...);
}
```